daemon/cmd/daemon_main.go

```diff
@@ -1365,9 +1370,11 @@ func runDaemon() {
 	bootstrapStats.enableConntrack.End(true)
 
 	bootstrapStats.k8sInit.Start()
-	// Wait only for certain caches, but not all!
-	// (Check Daemon.initK8sSubsystem() for more info)
-	<-d.k8sCachesSynced
+	if k8s.IsEnabled() {
+		// Wait only for certain caches, but not all!
+		// (Check Daemon.initK8sSubsystem() for more info)
+		<-d.k8sCachesSynced
+	}
```

Comment:
This change should be part of the previous commit. If the CI passes there is not need to re-trigger the CI after moving this change to the previous commit.